lib/ember-orbit/identity-map.js

```diff
@@ -0,0 +1,61 @@
+const { get } = Ember;
+
+export default Ember.Object.extend({
+  _schema: null,
+  _orbitCache: null,
+  _materialized: null,
+  _store: null,
+
+  init(...args) {
+    this._super.apply(this, ...args);
+
+    Ember.assert(this.get('_schema'), '_schema is required');
+    Ember.assert(this.get('_orbitCache'), '_orbitCache is required');
+
+    this.set('_materialized', {});
+  },
+
+  lookup(identifier) {
+    if(!identifier) return;
+
+    const {type, id} = identifier;
+
+    const materialized = this.get('_materialized');
+    const identifierKey = this._identifierKey(type, id);
+
+    return materialized[identifierKey] || this._materialize(type, id);
+  },
+
+  identifier(record) {
```

Comment:
These little methods for identifiers are littered all over the place, need to be consolidated.